### PR TITLE
Longest Common Subsequence in Python

### PR DESCRIPTION
**Pull request by Boot-Error**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [x] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [ ] I fixed #your-issue-number-here

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

**Comment by jrg94** [comment]
Hey @Boot-Error! I appreciate your contribution, but you'll need to make some changes if you want it included in the collection. Let me know if this is something you still want to work on. Otherwise, we'll have to close out this pull request.

**Comment by Boot-Error** [comment]
Hey,
Sorry for the delay

Yes I ll work on changes and get them pushed soon.

File: archive/p/python/lcs.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+def lcs(s1, s2):
+
+    if (len(s1) == 0 or len(s2) == 0): return []
+    elif (s1[-1] == s2[-1]): return lcs(s1[:-1], s2[:-1]) + [s1[-1]]
+    elif (s1[-1] != s2[-1]): return longest(lcs(s1, s2[:-1]), lcs(s1[:-1], s2))
+    else: pass
+
+if __name__ == "__main__":
+    
+    A = [1, 4, 5, 3, 15, 6]
```
Comment (jrg94) [inline archive/p/python/lcs.py]: Any reason these are initialized with arbitrary lists?

File: archive/p/python/README.md [hunk 2]
```diff
@@ -62,7 +63,7 @@ Welcome to Sample Programs in Python!
 - [ ] Simple Statements
   - [x] [Assignment Statement][5]
   - [ ] The `assert` Statement
-  - [ ] The `pass` Statement
+  - [ ] [The `pass` Statement][28]
```
Comment (jrg94) [inline archive/p/python/README.md]: Missing the check! ✔️

File: archive/p/python/README.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ Welcome to Sample Programs in Python!
 - [Game of Life in Python][4]
 - [Fibonacci Sequence in Python][24]
 - [Roman Numberal Conversion in Python][25]
+- [Longest Common Subsequence][]
```
Comment (jrg94) [inline archive/p/python/README.md]: Can you create an article issue like #32 and link it here.

File: archive/p/python/lcs.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
+
+import argparse
+
+def input_list(list_str):
+
+    return [int(x.strip(" "), 10) for x in list_str.split(',')]
```
Comment (jrg94) [inline archive/p/python/lcs.py]: By the way, I think this is really, really cool! I've used argparse in the past, but I didn't realize you could create your own input types. I was over hear looking through docs for `input_list` until I saw: 

```
type= can take any callable that takes a single string argument and returns the converted value:
```

Cool stuff!

**Pull request merged**